plugins/BEdita/Core/src/Filesystem/Thumbnail/GlideGenerator.php

```diff
@@ -54,19 +56,20 @@ class GlideGenerator extends ThumbnailGenerator
      * @param \BEdita\Core\Model\Entity\Stream $stream Stream entity instance.
      * @param array $options Thumbnail options.
      * @return string
+     * @throws \Cake\Http\Exception\BadRequestException
      */
     protected function getFilename(Stream $stream, array $options = [])
     {
-        $ext = pathinfo($stream->file_name, PATHINFO_EXTENSION);
-        if ($ext) {
-            $ext = '.' . $ext;
+        $ext = Hash::get($options, 'fm', 'jpg'); // jpg default
+        if (!in_array($ext, ['jpg', 'pjpg', 'png', 'gif', 'webp', 'avif'])) {
+            throw new BadRequestException(__d('bedita', 'Invalid thumbnail format: {0}', $ext));
```

Comment:
Where possible, I'd avoid using these HTTP-specific exceptions outside Controllers, Middlewares and Components.

I'd rather use the SPL `\InvalidArgumentException`, which fits the purpose better in my opinion. On the other hand, this would cause a **500 Internal Server Error** if uncaught, whereas the `\Cake\Http\Exception\BadRequestException` would cause a **400 Bad Request**.

Therefore, if a **400** is desired, one would need to catch this exception in the controller action, and re-throw the desired HTTP-specific one.

Reply:
What about the new `InvalidDataException` for now? Still not optimal, but maybe we can close this issue. @batopa @fquffio

Reply:
Seems the best option to me! 👍🏼